services/device-registry/src/main/java/org/eclipse/hono/service/deviceregistry/impl/ApplicationConfig.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.deviceregistry.impl;
+
+import org.springframework.beans.factory.config.ObjectFactoryCreatingFactoryBean;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import io.vertx.core.Vertx;
+
+/**
+ * Spring Boot configuration for the simple device registry server application.
+ *
+ */
+@Configuration
+public class ApplicationConfig {
+
+    private static final Vertx vertx = Vertx.vertx();
+
+    /**
+     * Gets the singleton Vert.x instance to be used by Hono.
+     * 
+     * @return the instance.
+     */
+    @Bean
+    public Vertx vertx() {
+        return vertx;
+    }
+
+    /**
+     * Exposes a factory for creating service instances as a Spring bean.
+     * 
+     * @return The factory.
+     */
+    @Bean
+    public ObjectFactoryCreatingFactoryBean deviceRegistryServerFactory() {
+        ObjectFactoryCreatingFactoryBean factory = new ObjectFactoryCreatingFactoryBean();
+        factory.setTargetBeanName(SimpleDeviceRegistryServer.class.getName());
+        return factory;
+    }
+
+    /**
+     * Exposes this service's configuration properties as a Spring bean.
+     * 
+     * @return The properties.
+     */
+    @Bean
+    @ConfigurationProperties(prefix = "hono.deviceRegistry")
```

Comment:
IMHO the prefix should not contain camel case segments, i.e. use `hono.device.registry` instead.